wrangler-service/src/main/java/co/cask/wrangler/service/common/AbstractWranglerHandler.java

```diff
@@ -198,6 +199,13 @@ protected Connection getValidatedConnection(ConnectionStore store, NamespacedId
       responder.sendJson(e.getCode(), new ServiceResponse<>(e.getMessage()));
     } catch (JsonSyntaxException e) {
       responder.sendJson(HttpURLConnection.HTTP_BAD_REQUEST, new ServiceResponse<Void>(e.getMessage()));
+    } catch (TransactionException e) {
+      // in case one of the handlers did not propagate exceptions within a transaction correctly.
+      if (e.getCause() != null) {
```

Comment:
When would cause be null?

Reply:
not sure, maybe if there is an actual transaction problem when talking to the datastore. Haven't actually seen a case myself

Reply:
Should not add this check then? I think if this happens, we should fix the handler/whoever is throwing the exception to propagate correct exception rather than handling it here. What do you think?

Reply:
I'll remove this catch clause